spec.html

```diff
@@ -28276,130 +28172,88 @@ <h1>Math.imul ( _x_, _y_ )</h1>
-            If _x_ is *-0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is 1, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ &lt; 0, return *NaN*.
+          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
+          1. If _n_ is 1, return *+0*.
+          1. If _n_ is *+&infin;*, return *+&infin;*.
+          1. Return an implementation-approximated value representing the result of the base 10 logarithm of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.log2">
         <h1>Math.log2 ( _x_ )</h1>
         <p>Returns an implementation-approximated value representing the result of the base 2 logarithm of _x_.</p>
-        <ul>
-          <li>
-            If _x_ is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is less than 0, the result is *NaN*.
-          </li>
-          <li>
-            If _x_ is *+0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is *-0*, the result is *-&infin;*.
-          </li>
-          <li>
-            If _x_ is 1, the result is *+0*.
-          </li>
-          <li>
-            If _x_ is *+&infin;*, the result is *+&infin;*.
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN* or _n_ &lt; 0, return *NaN*.
+          1. If _n_ is *+0* or _n_ is *-0*, return *-&infin;*.
+          1. If _n_ is 1, return *+0*.
+          1. If _n_ is *+&infin;*, return *+&infin;*.
+          1. Return an implementation-approximated value representing the result of the base 2 logarithm of _n_.
+        </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.max">
-        <h1>Math.max ( _value1_, _value2_, ..._values_ )</h1>
+        <h1>Math.max ( ..._values_ )</h1>
         <p>Given zero or more arguments, calls ToNumber on each of the arguments and returns the largest of the resulting values.</p>
-        <ul>
-          <li>
-            If no arguments are given, the result is *-&infin;*.
-          </li>
-          <li>
-            If any value is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            The comparison of values to determine the largest value is done using the Abstract Relational Comparison algorithm except that *+0* is considered to be larger than *-0*.
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _values_ be a List whose elements are, in left to right order, the arguments that were passed to this function invocation.
+          1. Let _len_ be the number of elements in _values_.
+          1. Let _v_ be *-&infin;*.
+          1. Repeat, while _values_ is not empty,
+            1. Remove the first element from _values_ and let _value_ be the value of the element.
```

Comment:
I would just make this two steps.

```suggestion
            1. Let _value_ be the first element of _values_.
            1. Remove the first element from _values_.
```

Reply:
This format is also used in `Array` methods (e.g. `Array.prototype.concat`) which is from where I borrowed it.

Reply:
Thanks, I've sent PR https://github.com/tc39/ecma262/pull/2137 to address those cases.